priority/priority.erl

```diff
@@ -1,47 +1,45 @@
 -module(priority).
+
 -export([start/0,
          send_vip/2,
          send_normal/2]).
- -include_lib("eunit/include/eunit.hrl").
 
-priority_process_msg(Msg) ->
-    io:format("Processing importante msg: ~p~n",[Msg]),
-    timer:sleep(500). % only for being able to produce faster than cusume (testing...)
+-include_lib("eunit/include/eunit.hrl").
 
-priority_loop_vip(State) ->
-    receive 
-        {vip,Msg} ->
-            priority_process_msg(Msg),
-            priority_loop_vip(State)
-    after 0 ->
-            priority_loop(State)
-    end.
+start() ->
+    spawn(fun priority_loop/0).
 
-priority_loop(State) ->
-    receive 
-        {normal,{Pid,exit}} ->
-            Pid ! exit;
-        {normal,Msg} ->
-            io:format("normal msg : ~p~n",[Msg]),
-            priority_loop(State);
-        {vip,Msg}->
-            priority_process_msg(Msg),
-            priority_loop_vip(State);
-        {ready,Pid} ->
-            Pid ! read %end test
-    end.
+send_vip(Pid, Msg) ->
+    Pid ! {vip, Msg}.
 
-start() ->
-    spawn(fun() -> priority_loop(state_place_holder) end).
+send_normal(Pid, Msg) ->
+    Pid ! {normal, Msg}.
 
-send_vip(Pid,Msg) ->
-    Pid ! {vip,Msg}.
+process_vip_msg(Msg) ->
+    io:format("Processing importante msg: ~p~n", [Msg]),
+    timer:sleep(500). % only for being able to produce faster than cusume (testing...)
 
-send_normal(Pid,Msg) ->
-    Pid ! {normal,Msg}.
+%%% internal functions
+
+priority_loop() ->
+    receive
+        {vip, Msg} ->
+            process_vip_msg(Msg),
+            priority_loop()
+    after 0 ->
+            receive 
+                {normal, Msg} ->
+                    io:format("normal msg : ~p~n", [Msg]),
```

Comment:
still not processing normal messages. I'd say the processing function should be the same in both cases (you could pass the priority value if you want, to log it).